File: we_13_1_a_chi_squared_raim_detection.py
# Integrated Navigation Systems — Chapter 13: Integrity, Fault Detection, and ARAIM
# Worked Example 13.1 — A chi-squared RAIM detection
# Companion code (educational reference implementation; see the book for context).

from scipy.stats import chi2   # Td = chi2.ppf(0.95,3) = 7.815
SSE=18.7; T=SSE/1.0; Td=7.815
print('FAULT' if T>Td else 'OK')   # FAULT (18.7 > 7.81)

File: we_13_1_a_chi_squared_raim_detection.m
% Integrated Navigation Systems — Chapter 13: Integrity, Fault Detection, and ARAIM
% Worked Example 13.1 — A chi-squared RAIM detection
% Companion code (educational reference implementation; see the book for context).

SSE=18.7; sigma2=1.0; T=SSE/sigma2; Td=7.81;
fprintf('T=%.1f %s Td=%.2f -> %s\n', T, '>', Td, 'FAULT');   % 18.7 > 7.81
